internal/controller/operator/converter/apis_test.go

```diff
@@ -14,259 +14,243 @@ import (
-					EndpointRelabelings: vmv1beta1.EndpointRelabelings{
-						RelabelConfigs: []*vmv1beta1.RelabelConfig{
-							{
-								Action:       "drop",
-								SourceLabels: []string{"__meta__instance"},
-							},
+		},
+		want: []vmv1beta1.Endpoint{
+			{
+				EndpointScrapeParams: vmv1beta1.EndpointScrapeParams{
+					Path: "/metrics",
+				},
+				Port: "9100",
+				EndpointRelabelings: vmv1beta1.EndpointRelabelings{
+					RelabelConfigs: []*vmv1beta1.RelabelConfig{
+						{
+							Action:       "drop",
+							SourceLabels: []string{"__meta__instance"},
 						},
 					},
 				},
 			},
 		},
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if got := convertEndpoint(tt.args.promEndpoint); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("ConvertEndpoint() \ngot:  \n%v\n, \nwant: \n%v", got, tt.want)
-			}
-		})
-	}
+	f(o)
 }
 
 func TestConvertServiceMonitor(t *testing.T) {
-	type args struct {
-		serviceMon *promv1.ServiceMonitor
-	}
-	tests := []struct {
-		name string
-		args args
+	type opts struct {
+		cfg  *promv1.ServiceMonitor
 		want vmv1beta1.VMServiceScrape
-	}{
-		{
-			name: "with metricsRelabelConfig",
-			args: args{
-				serviceMon: &promv1.ServiceMonitor{
-					Spec: promv1.ServiceMonitorSpec{
-						Endpoints: []promv1.Endpoint{
+	}
+	f := func(opts opts) {
+		t.Helper()
+		got := ConvertServiceMonitor(opts.cfg, &config.BaseOperatorConf{
+			FilterPrometheusConverterLabelPrefixes:      []string{"app.kubernetes", "helm.sh"},
+			FilterPrometheusConverterAnnotationPrefixes: []string{"another-annotation-filter", "app.kubernetes"},
+		})
+		if !reflect.DeepEqual(*got, opts.want) {
+			t.Errorf("ConvertServiceMonitor() got = \n%v, \nwant \n%v", got, opts.want)
+		}
+	}
+
+	// with metricsRelabelConfig
+	o := opts{
+		cfg: &promv1.ServiceMonitor{
+			Spec: promv1.ServiceMonitorSpec{
+				Endpoints: []promv1.Endpoint{
+					{
+						MetricRelabelConfigs: []promv1.RelabelConfig{
 							{
-								MetricRelabelConfigs: []promv1.RelabelConfig{
-									{
-										Action:       "drop",
-										SourceLabels: []promv1.LabelName{"__meta__instance"},
-									},
-								},
+								Action:       "drop",
+								SourceLabels: []promv1.LabelName{"__meta__instance"},
 							},
 						},
 					},
 				},
 			},
-			want: vmv1beta1.VMServiceScrape{
-				Spec: vmv1beta1.VMServiceScrapeSpec{
-					Endpoints: []vmv1beta1.Endpoint{
-						{
-							EndpointRelabelings: vmv1beta1.EndpointRelabelings{
-								MetricRelabelConfigs: []*vmv1beta1.RelabelConfig{
-									{
-										Action:       "drop",
-										SourceLabels: []string{"__meta__instance"},
-									},
+		},
+		want: vmv1beta1.VMServiceScrape{
+			Spec: vmv1beta1.VMServiceScrapeSpec{
+				Endpoints: []vmv1beta1.Endpoint{
+					{
+						EndpointRelabelings: vmv1beta1.EndpointRelabelings{
+							MetricRelabelConfigs: []*vmv1beta1.RelabelConfig{
+								{
+									Action:       "drop",
+									SourceLabels: []string{"__meta__instance"},
```

Comment:
```suggestion
									SourceLabels: []promv1.LabelName{"__meta__instance"},
```
nit: we can replace them all

Reply:
this package is not in diff already